### PR TITLE
multiprocessing.pool: fix __init__ methods

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: stdlib/multiprocessing/pool.pyi [hunk 1]
```diff
@@ -11,10 +11,7 @@ _T = TypeVar("_T")
 
 class ApplyResult(Generic[_T]):
     def __init__(
-        self,
-        pool: Pool,
-        callback: Optional[Callable[[_T], None]] = ...,
-        error_callback: Optional[Callable[[BaseException], None]] = ...,
+        self, pool: Pool, callback: Optional[Callable[[_T], None]], error_callback: Optional[Callable[[BaseException], None]]
```
Comment (srittau) [inline stdlib/multiprocessing/pool.pyi:14]: At least in Python 3.6, the first argument is called `cache`.

File: stdlib/multiprocessing/pool.pyi [hunk 2]
```diff
@@ -26,9 +23,18 @@ class ApplyResult(Generic[_T]):
...
-class MapResult(ApplyResult[List[_T]]): ...
+class MapResult(ApplyResult[List[_T]]):
+    def __init__(
+        self,
+        pool: Pool,
+        chunksize: int,
+        length: int,
+        callback: Optional[Callable[[List[_T]], None]],
+        error_callback: Optional[Callable[[BaseException], None]],
+    ) -> None: ...
```
Comment (srittau) [inline stdlib/multiprocessing/pool.pyi:34]: Same here. Also, please use `| None` and built-in generics, at least for new annotations.

File: stdlib/multiprocessing/pool.pyi [hunk 3]
```diff
@@ -26,9 +23,18 @@ class ApplyResult(Generic[_T]):
...
 
 class IMapIterator(Iterator[_T]):
+    def __init__(self, pool: Pool) -> None: ...
```
Comment (srittau) [inline stdlib/multiprocessing/pool.pyi:37]: Here too.

**Review by srittau — CHANGES_REQUESTED** [review]

File: stdlib/multiprocessing/pool.pyi (hunk 1 above)
Comment (hauntsaninja) [reply stdlib/multiprocessing/pool.pyi:14]: Yeah, in 3.6 and 3.7 the cache argument is a multiprocessing.pool._PoolCache, changed in bpo-35378. We don't have typing for that, so I just moved around the stubtest allowlists instead. But you called me out on my laziness, so I'll get around to adding all the branches for 3.7 :-)

Comment (srittau) [reply stdlib/multiprocessing/pool.pyi:14]: Thanks! And feel free to merge when done.

**Comment by thetorpedodog** [comment]
Thank you for getting this fix in!

**Pull request merged**